src/pynwb/tests/test_example_usage_all.py

```diff
@@ -0,0 +1,242 @@
+
+    pm = ProbeModel(
+        name="Neuropixels 1.0",
+        description="A neuropixels probe",
+        model="Neuropixels 1.0",
+        manufacturer="IMEC",
+        planar_contour_in_um=[[-10.0, -10.0], [10.0, -10.0], [10.0, 10.0], [-10.0, 10.0]],
+        contacts_table=contacts_table,
+    )
+    # TODO put this into /general/device_models
+    nwbfile.add_device(pm)
+
+    probe = Probe(
+        name="Neuropixels Probe 1",
+        identifier="28948291",
+        probe_model=pm,
+    )
+    nwbfile.add_device(probe)
+
+    pi = ProbeInsertion(
+        position_reference="(AP, ML, DV) = (0, 0, 0) corresponds to bregma at the cortical surface.",
+        hemisphere="left",
+        depth_in_mm=10.0,
+        insertion_position_ap_in_mm=2.0,
+        insertion_position_ml_in_mm=-4.0,
+        insertion_angle_roll_in_deg=-10.0,
+        insertion_angle_pitch_in_deg=0.0,
+        insertion_angle_yaw_in_deg=0.0,
+    )
+
+    channels_table = ChannelsTable(
+        name="Neuropixels1ChannelsTable",  # test custom name
+        description="Test channels table",
+        reference_mode="Referenced to channel 2.",
+        position_reference="(AP, ML, DV) = (0, 0, 0) corresponds to bregma at the cortical surface.",
+        position_confirmation_method="Histology",
+        probe=probe,
+        probe_insertion=pi,
+        target_tables={
+            "contact": probe.probe_model.contacts_table,
+            "reference_contact": probe.probe_model.contacts_table,
+        },
+        # TODO should not need to specify the above
+    )
+
+    # all of the keyword arguments in add_row are optional
+    channels_table.add_row(
+        contact=0,
+        reference_contact=2,
+        filter="High-pass at 300 Hz",
+        estimated_position_ap_in_mm=2.0,
+        estimated_position_ml_in_mm=-5.0,
+        estimated_position_dv_in_mm=-9.5,
+        estimated_brain_area="CA3",
+        confirmed_position_ap_in_mm=2.0,
+        confirmed_position_ml_in_mm=-4.9,
+        confirmed_position_dv_in_mm=-9.5,
+        confirmed_brain_area="CA3",
+    )
+    channels_table.add_row(
+        contact=1,
+        reference_contact=2,
+        filter="High-pass at 300 Hz",
+        estimated_position_ap_in_mm=2.0,
+        estimated_position_ml_in_mm=-4.9,
+        estimated_position_dv_in_mm=-9.3,
+        estimated_brain_area="CA3",
+        confirmed_position_ap_in_mm=2.0,
+        confirmed_position_ml_in_mm=-4.8,
+        confirmed_position_dv_in_mm=-9.3,
+        confirmed_brain_area="CA3",
+    )
+
+    # put this in nwbfile.acquisition for testing
+    nwbfile.add_acquisition(channels_table)
+
+    channels = DynamicTableRegion(
+        name="channels",  # NOTE: this must be named "channels" when used in ExtracellularSeries
+        data=[0, 1, 2],
+        description="All of the channels",
+        table=channels_table,
+    )
+
+    es = ExtracellularSeries(
+        name="ExtracellularSeries",
+        data=[0.0, 1.0, 2.0],
```

Comment:
I am confused here. If I read well this represents data with one channel. However, you adding as a dataset a region that points to three channels. Is that right? Should it not throw an error because the region is pointing to does not match the number of channels that the ExtracellularSeries represents?

Reply:
Good catch! Yes, you are right.